### PR TITLE
Fix 65000 character limit when publishing to github API

### PR DESCRIPTION
**Pull request by jsclifford**
# Summary of PR changes
* Check character limit of test outcome if under 65535 count.
* Output to file if hit that limit.

## PR Requirements
- [x] For major, minor, or breaking changes, at least one of the commit messages contains the appropriate `+semver:` keywords.
  - See the *Incrementing the Version* section of the repository's README.md for more details.
- [x] The action code does not contain sensitive information.

*NOTE: If the repo's workflow could not automatically update the `README.md`, it should be updated manually with the next version.  For javascript actions, if the repo's workflow could not automatically recompile the action it should also be updated manually as part of the PR.*

### CODE REVIEW

**Comment by jsclifford** [comment]
Successful workflow run with this action version.  
https://github.com/im-enrollment/shopping/actions/runs/5180783371/jobs/9337378262#step:11:1
Status check: https://github.com/im-enrollment/shopping/actions/runs/5180783371/jobs/9337392023

File: src/main.js [hunk 1]
```diff
@@ -26,7 +26,34 @@ async function run() {
...
+      );
+      const outcomeFile = './test-outcome.md';
+      fullMarkupData = markupData;
+      markupData = getMarkupForJson(resultsJson, reportName, true);
+
+      const outcomeFilePath = createOutcomeFile(outcomeFile, fullMarkupData);
+      core.setOutput('test-outcome-truncated', 'true');
+      core.setOutput('test-outcome-file-path', outcomeFilePath);
+
+      if (markupData.length > 65535) {
```
Comment (mike-schenk) [inline src/main.js]:
```suggestion
      if (markupData.length > characterLimit) {
```

**Review by mike-schenk — COMMENTED** [review]
I've been racking my brain to come up with a better way to limit the length rather than producing the Markdown twice... and then still possibly truncating the final result. I haven't come up with anything.

**Comment by jsclifford** [comment]
> I've been racking my brain to come up with a better way to limit the length rather than producing the Markdown twice... and then still possibly truncating the final result. I haven't come up with anything.

Yes.  It is a complicated problem.  It takes so little amount of time though I compromised to creating two markdown strings. That is why I try to reduce the size the first time, then just cut it off if it is still too big.

**Review by mike-schenk — APPROVED** [review]

**Pull request merged**